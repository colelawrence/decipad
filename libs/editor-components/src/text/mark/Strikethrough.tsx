import { atoms } from '@decipad/ui';
import { PlateComponent } from '@decipad/editor-types';

export const Strikethrough: PlateComponent = ({ attributes, children }) => {
  return (
    <span {...attributes}>
      <atoms.Strikethrough>{children}</atoms.Strikethrough>
    </span>
  );
};
