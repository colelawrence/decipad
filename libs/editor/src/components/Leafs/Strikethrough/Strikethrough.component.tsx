import styled from '@emotion/styled';
import { SlatePluginComponent } from '@udecode/slate-plugins';
import React from 'react';

const Leaf = styled('span')({
  textDecoration: 'line-through',
});

export const Strikethrough: SlatePluginComponent = ({
  attributes,
  children,
}) => {
  return <Leaf {...attributes}>{children}</Leaf>;
};
