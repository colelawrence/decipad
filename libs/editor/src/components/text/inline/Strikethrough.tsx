import { atoms } from '@decipad/ui';
import { PlatePluginComponent } from '@udecode/plate';

export const Strikethrough: PlatePluginComponent = ({
  attributes,
  children,
}) => {
  if (!('data-slate-leaf' in attributes)) {
    throw new Error('Strikethrough is a leaf');
  }

  return (
    <atoms.Strikethrough slateAttrs={attributes}>
      {children}
    </atoms.Strikethrough>
  );
};
