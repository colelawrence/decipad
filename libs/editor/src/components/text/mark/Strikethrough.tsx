import { atoms } from '@decipad/ui';
import { PlateComponent } from '../../../types';

export const Strikethrough: PlateComponent = ({ attributes, children }) => {
  return (
    <span {...attributes}>
      <atoms.Strikethrough>{children}</atoms.Strikethrough>
    </span>
  );
};
