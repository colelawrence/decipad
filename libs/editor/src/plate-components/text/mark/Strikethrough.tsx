import { types } from '@decipad/editor-config';
import { atoms } from '@decipad/ui';

export const Strikethrough: types.PlateComponent = ({
  attributes,
  children,
}) => {
  return (
    <span {...attributes}>
      <atoms.Strikethrough>{children}</atoms.Strikethrough>
    </span>
  );
};
