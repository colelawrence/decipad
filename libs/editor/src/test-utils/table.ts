import {
  ELEMENT_TABLE,
  ELEMENT_TH,
  ELEMENT_TR,
  ELEMENT_TD,
} from '@udecode/plate';
import {
  InteractiveTable,
  TableCellType,
  Td,
  Th,
} from '../plugins/InteractiveTable/table';
import {
  ELEMENT_TABLE_CAPTION,
  ELEMENT_THEAD,
  ELEMENT_HEAD_TR,
  ELEMENT_TBODY,
} from '../utils/elementTypes';

/**
 * Creates a row with 1 named column and {@param numberOfRows} rows of non-empty cells.
 */
export const createTable = (numberOfRows: number): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType: 'string' },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: Array.from({ length: numberOfRows }, (_, i) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType: 'string' },
            children: [{ text: `Col 1 Row ${i + 1}` }],
          },
        ],
      })),
    },
  ],
});

export const createTableWithTypeAndData = (
  cellType: TableCellType,
  data: string[]
): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: data.map((text) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType },
            children: [{ text }],
          },
        ],
      })),
    },
  ],
});

export const printTable = ({
  children: [_, thead, tbody],
}: InteractiveTable): string[] => {
  const tdsToStrings = (tds: (Th | Td)[]) =>
    tds.map((td) => td.children[0].text).join(' | ');

  return [
    tdsToStrings(thead.children[0].children),
    ...tbody.children.map((tr) => tdsToStrings(tr.children)),
  ];
};
