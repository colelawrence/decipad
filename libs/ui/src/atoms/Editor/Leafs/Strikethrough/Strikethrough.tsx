import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';

const styles = css({
  textDecoration: 'line-through',
});

export const StrikethroughLeaf: PlatePluginComponent = ({
  attributes,
  children,
}) => {
  return (
    <span css={styles} {...attributes}>
      {children}
    </span>
  );
};
