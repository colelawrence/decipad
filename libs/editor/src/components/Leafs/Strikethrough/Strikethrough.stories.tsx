import { Strikethrough } from './Strikethrough.component';

export default {
  title: 'Editor/Leafs/Strikethrough',
  component: Strikethrough,
};

export const Default = (args: { text: string }) => (
  <Strikethrough
    leaf={{ attributes: {}, text: '' }}
    element={{ children: [{ text: '' }] }}
    text={{ text: '' }}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
      'data-slate-leaf': true,
    }}
  >
    {args.text}
  </Strikethrough>
);

Default.args = {
  text: 'Strikethrough',
};

Default.storyName = 'Strikethrough';
