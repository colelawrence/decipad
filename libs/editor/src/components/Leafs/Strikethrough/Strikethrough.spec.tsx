import React from 'react';
import { render } from 'test-utils';
import { Default } from './Strikethrough.stories';

describe('Strikethrough Leaf', () => {
  it('matches snapshot', () => {
    const { container } = render(<Default text="Strikethrough" />);
    expect(container).toMatchSnapshot();
  });
});
