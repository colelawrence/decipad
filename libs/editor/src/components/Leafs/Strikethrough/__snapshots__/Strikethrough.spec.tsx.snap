// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Strikethrough Leaf matches snapshot 1`] = `
.emotion-0 {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

<div>
  <span
    class="emotion-0"
    data-slate-leaf="true"
    data-slate-node="element"
  >
    Strikethrough
  </span>
</div>
`;
